## Data Analyst Interview Assignment

### 1. Data Load

In [7]:
# Libraries
import pandas as pd

In [8]:
file_path = 'Data Analyst Interview Assignment 2024 edition.xlsx'
df = pd.read_excel(file_path)

In [9]:
df.head()

Context
0  The "Raw Data" tab contains detailed informati...
1                                                NaN
2                                         Assignment
3  1. Please find the total Valid Bet Amount in SGD?
4  2. From your observations of the data, please ...

These seems to be the task requirements, therefore the data must be in another sheet.

In [10]:
xls = pd.ExcelFile(file_path)
sheet_names = xls.sheet_names
sheet_names

['Assignment', 'Raw Data', 'Definition']

I'm going to read the metadata first

In [11]:
df_third_sheet = pd.read_excel(file_path, sheet_name=sheet_names[2])
df_third_sheet.head(20)

Variable     Type  \
0   Transaction Id      INT   
1             Odds    FLOAT   
2           Status      STR   
3          Winlost    FLOAT   
4    LiveIndicator      INT   
5          Betteam  General   
6      Winlostdate     Time   
7       ActualRate    FLOAT   
8          MatchId      INT   
9          Bettype      INT   
10    Actual_Stake    FLOAT   
11        Currency      STR   
12        LeagueId      INT   
13       TransDate     Time   
14      WebSite Id      INT   
15     leagueGroup      STR   
16          MaxBet    FLOAT   
17     DangerLevel      INT   

                                           Definition  \
0       The identification number of each transaction   
1                                        Betting Odds   
2                          Status of the player's bet   
3   The amount that the player won or lost (includ...   
4   Whether the bet placed was before the match ki...   
5   What was the bet option placed, in conjunction...   
6                                          Event Date   
7         Exchange rate of each local currency to SGD   
8                Game ID (indicates a specific match)   
9   What was the type of bet placed, in conjunctio...   
10                                Player's bet amount   
11                  Local currency of the transaction   
12            League ID (indicates a specific League)   
13                The time when the player placed bet   
14                                    Customer Number   
15                                        League size   
16    Maximum amount of stake that a player can place   
17                                        Player Risk   

                                               Remark  
0                      Every transaction ID is unique  
1                                                 NaN  
2   The status (won, lose, draw) will be considere...  
3                 This value is in the local currency  
4                               0: Non Live ; 1: Live  
5   h:Home Team ; a:Away Team (h:Over ; a:Under) ;...  
6                                                 NaN  
7                                                 NaN  
8                                                 NaN  
9   1: Handicap ; 2: Odd/Even ; 3: Over/Under ; 4:...  
10                This value is in the local currency  
11                                                NaN  
12  3: English Premier League ; 4: Italy serie A ;...  
13                                                NaN  
14              One webSite Id indicates one Customer  
15                                                NaN  
16                This value is in the local currency  
17         0 (No risk player) - 70 (high risk player)

### 2. Exploratory Data Analysis

In [12]:
df_second_sheet = pd.read_excel(file_path, sheet_name=sheet_names[1])
df_second_sheet.head()

Transaction Id  Odds Status  Winlost  LiveIndicator   Betteam Winlostdate  \
0       101063377 -0.31    WON    393.0              1         h  2023-02-28   
1       101063392 -1.17    WON    108.5              1         h  2023-02-28   
2       101063388  6.76   LOSE      0.0              1  00:01:00  2023-02-28   
3        12685510  0.86    WON     37.2              1         h  2023-02-28   
4        12685513  1.42    WON    242.0              1         h  2023-02-28   

   ActualRate  MatchId  Bettype  Actual_Stake Currency  LeagueId  \
0     0.03775  5119849        9          93.0      THB         4   
1     0.08200  5119849        1          58.5      IDR         4   
2     0.08200  5119849        4          13.0      IDR         4   
3     0.03775  5119849        1          20.0      THB         4   
4     0.03775  5119849        9         100.0      THB         4   

         Transaction Date  WebSite Id leagueGroup    MaxBet  DangerLevel  
0 2023-02-28 16:22:08.540          97         BIG   15000.0            0  
1 2023-02-28 16:22:17.883        1171         BIG   25000.0           10  
2 2023-02-28 16:22:15.537        1171         BIG   10000.0            0  
3 2023-02-28 16:22:31.700           0         BIG  500000.0            0  
4 2023-02-28 16:22:47.747           0         BIG   20000.0            0

Get a sample to get an idea of the dataframe beyond the top rows.

In [43]:
df_second_sheet.sample(10)

Transaction Id  Odds Status  Winlost  LiveIndicator Betteam  \
478615         10043160 -0.61    WON   805.00              0       a   
387500          8933176  1.26    WON    66.83              1       h   
176612         12646640  0.98   DRAW   101.00              1       a   
571456         35883001  1.28   LOSE     0.00              1       h   
133658         36270096 -0.91    WON    95.50              0       h   
172976         42441664 -1.57    WON   321.25              1       a   
57145         100846142  1.56   LOSE     0.00              1       h   
821286         34730892  1.40    WON   720.00              0       h   
870319         35710256 -2.64   LOSE     0.00              1       h   
1013066        35451070  4.90   LOSE     0.00              0       1   

        Winlostdate  ActualRate  MatchId  Bettype  Actual_Stake Currency  \
478615   2023-02-21        0.05  4839770        1        305.00      VND   
387500   2023-02-25        0.08  5114289        1         41.00      IDR   
176612   2023-02-26        0.31  5113440        7        101.00      MYR   
571456   2023-02-19        0.08  5091012        1         25.00      IDR   
133658   2023-02-26        0.04  5067715        1         45.50      THB   
172976   2023-02-26        0.08  5113434        3        196.25      IDR   
57145    2023-02-27        0.08  5143217        1         39.00      IDR   
821286   2023-02-18        0.04  5108170        1        300.00      THB   
870319   2023-02-18        0.08  5096739        1        398.64      IDR   
1013066  2023-02-17        0.08  5096738        5         50.00      IDR   

         LeagueId        Transaction Date  WebSite Id leagueGroup    MaxBet  \
478615         13 2023-02-21 13:08:57.223           0         BIG  57692.00   
387500          5 2023-02-25 11:29:52.063           0         BIG  30000.00   
176612          8 2023-02-26 10:33:25.003           0         BIG  16129.00   
571456          8 2023-02-19 09:00:29.710       40001         BIG   5000.00   
133658          3 2023-02-26 08:20:43.603       10034         BIG  50000.00   
172976          8 2023-02-26 10:24:10.603           0         BIG  30000.00   
57145           6 2023-02-27 17:20:17.637          20         BIG 150000.00   
821286          5 2023-02-18 10:25:14.893       10034         BIG  50000.00   
870319          4 2023-02-18 11:42:05.407       40001         BIG   5000.00   
1013066         4 2023-02-17 01:50:57.850       40001         BIG  18761.00   

         DangerLevel  AmountInSGD  AmmountWinlost  
478615            10        15.86           41.86  
387500             0         3.36            5.48  
176612             0        31.31           31.31  
571456            30         2.05            0.00  
133658            20         1.72            3.61  
172976             0        16.09           26.34  
57145              0         3.20            0.00  
821286             0        11.32           27.18  
870319            30        32.69            0.00  
1013066            0         4.10            0.00

In [14]:
df_second_sheet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 18 columns):
 #   Column            Non-Null Count    Dtype         
---  ------            --------------    -----         
 0   Transaction Id    1048575 non-null  int64         
 1   Odds              1048575 non-null  float64       
 2   Status            1048575 non-null  object        
 3   Winlost           1048575 non-null  float64       
 4   LiveIndicator     1048575 non-null  int64         
 5   Betteam           1048575 non-null  object        
 6   Winlostdate       1048575 non-null  datetime64[ns]
 7   ActualRate        1048575 non-null  float64       
 8   MatchId           1048575 non-null  int64         
 9   Bettype           1048575 non-null  int64         
 10  Actual_Stake      1048575 non-null  float64       
 11  Currency          1048575 non-null  object        
 12  LeagueId          1048575 non-null  int64         
 13  Transaction Date  1048575 non-null  dateti

There are no missing values, so aritmethic operations are safe to be done.

### Task 1. Please find the total Valid Bet Amount in SGD?


First, lest check if there are any bets in the same target currency (SGD is the target)

In [16]:
df_second_sheet['Currency'].value_counts()

Currency
IDR       605047
THB       313312
VND        59139
MYR        32480
Others     14017
USD         8735
CNY         8525
KRW         7320
Name: count, dtype: int64

There is no SGD currency, therefore all records must be converted

In [18]:
# Convert the stake amount with the rate
df_second_sheet['AmountInSGD'] = df_second_sheet['Actual_Stake'] * df_second_sheet['ActualRate']

# Get the total sum with two decimal points
df_second_sheet['AmountInSGD'].sum().round(2)

50162771.37

Answer: The total valid amount in SGD is 50,162,771.37

### Task 2. From your observations of the data, please share any insights that you have.


The data corresponds to bets from a different Asian countries. From the metadata, the transactions belong to a website or online platform.

In [22]:
# Set option to display large numbers
pd.set_option('display.float_format', lambda x: '%.2f' % x)

In [23]:
# Groups by currency and sums by total SGD and order by value
df_second_sheet.groupby('Currency')['AmountInSGD'].sum().reset_index().sort_values(by='AmountInSGD', ascending=False)

Currency  AmountInSGD
1      IDR  14628323.18
5      THB  14495059.87
7      VND   7550311.81
3      MYR   5280124.40
2      KRW   3866209.85
6      USD   1870787.52
0      CNY   1423556.28
4   Others   1048398.46

From the output, the top 3 countries that bet more ammount of money are Indonesia, THailand, and Vietnam.

In [28]:
# Convert the win/lost amount with the rate
df_second_sheet['AmmountWinlost'] = df_second_sheet['Winlost'] * df_second_sheet['ActualRate']
# Groups by currency and sums by total SGD and order by value
df_second_sheet.groupby('Currency')['Winlost'].sum().reset_index().sort_values(by='Winlost', ascending=True)

Currency       Winlost
6      USD    1350091.40
0      CNY    8061321.97
3      MYR   16923562.29
7      VND  149701895.42
1      IDR  179546794.46
5      THB  395915786.85
4   Others 3260462620.74
2      KRW 3308677543.18

From the output, the top 3 earner countries are KOrea, other countries, and Thailand


In [27]:
# COunt the number of transactions
df_second_sheet.groupby('Currency')['Transaction Id'].count().reset_index().sort_values(by='Transaction Id', ascending=False)

Currency  Transaction Id
1      IDR          605047
5      THB          313312
7      VND           59139
3      MYR           32480
4   Others           14017
6      USD            8735
0      CNY            8525
2      KRW            7320

From the output, the top 3 betting countries by transactions are Indonesia THailand, and Vietnam


### Task 3. From the insights that you have, please make recommendations on how to improve winlost from a website operator's point of view.

From the insights,
- the operator has his best clients in Indonesia, Thailand, and Vietnam. Therefore, the operator can improve the website's language to give a better experience to this countries to keep the best stream of income.
- the operator had suffer fewer losses from bets in the US and China. Therefore, the operator can prompt the customers from these countries to take more risk, to bet more.
- the operator can benefit from increasing the transactions fees from the large volume of operations in Indonesia THailand, and Vietnam. Even if the fee is small, the amount of transactions would mean a large profit.